Clang-format check for itt collector

ilyachur opened the pull request:
### Details:
 - Enabled clang-format for ITT collector lib instead of cpplint

alalek reviewed (commented):
BTW, It is a bad practice to "format" 3rdparty libraries in general. This would broke "easy and clear" upgrading / merging of upstream patches.

Consider excluding these paths from the linter checks instead.

ilyachur commented:
> BTW, It is a bad practice to "format" 3rdparty libraries in general. This would broke "easy and clear" upgrading / merging of upstream patches.
> 
> Consider excluding these paths from the linter checks instead.

I understand that it is a bad practice to apply custom code style for thirdparty libraries. 

But here we have an exception. 

First of all as you can see we already had an enabled `cpplint` checker for this component. I just enabled `clang-format` for this component.

At the second moment in any case we have to always manually upgrade this component by two reasons:

- We took only several parts of original component and removed majority features, it means that each time we need to understand which parts are needed for us.
- We changed the code to fix SDL for this component
